internal/commandhandler/commandhandler.go

```diff
@@ -555,3 +567,43 @@ func (h *BaseCommandHandler) RespAuth(args []string) interface{} {
 
 	return clientio.OK
 }
+
+func GetWALReplayHandler(shardManager *shard.ShardManager) (*BaseCommandHandler, error) {
+	// Create channels for the replay handler
+	responseChan := make(chan *ops.StoreResponse)
+	preprocessingChan := make(chan *ops.StoreResponse)
+	watchSubscriptionChan := make(chan watchmanager.WatchSubscription)
+	globalErrorChan := make(chan error)
+	ioThreadReadChan := make(chan []byte)
+	ioThreadWriteChan := make(chan interface{})
+	ioThreadErrChan := make(chan error)
+
+	// Create a new command handler for WAL replay
+	replayHandler := NewCommandHandler(
+		"wal-replay",
+		responseChan,
+		preprocessingChan,
+		watchSubscriptionChan,
+		nil, // No parser needed for replay
+		shardManager,
+		globalErrorChan,
+		ioThreadReadChan,
+		ioThreadWriteChan,
+		ioThreadErrChan,
+		nil,  // No WAL needed for replay handler
+		true, // isReplayHandler
+	)
+
+	// Register the command handler with the shard manager
+	shardManager.RegisterCommandHandler(replayHandler.ID(), responseChan, preprocessingChan)
+
+	// Start the command handler in a background goroutine
+	ctx := context.Background()
+	go func() {
+		if err := replayHandler.Start(ctx); err != nil {
+			slog.Error("WAL replay handler failed", slog.Any("error", err))
+		}
+	}()
```

Comment:
Why does this need to happen in a thread? Ideally, DiceDB should not start accepting and serving the requests until the entire WAL is replayed on boot.